src/flamegraph/merge.rs

```diff
@@ -115,12 +115,23 @@ where
     let mut delta = None;
     let mut delta_max = 1;
     let mut stripped_fractional_samples = false;
+    let mut prev_line = None;
+    let mut unsorted_detected = false;
     for line in lines {
         let mut line = line.trim();
         if line.is_empty() {
             continue;
         }
 
+        if !unsorted_detected {
+            if let Some(prev_line) = prev_line {
+                if prev_line > line {
+                    unsorted_detected = true;
+                    warn!("Unsorted input lines detected");
```

Comment:
I think we should just return if we detect unsorted input. The output will already be messed up.

Reply:
Good point.

Reply:
So should we just return the values we have up to this point then after logging an error?

Reply:
I think we should actually just return a full-out `Err`, since the results just won't be useful at that point.

Reply:
Ok, so you want to change the return type to a `Result`? What about the error type? We don't currently have a custom error type for flamegraph. `from_sorted_lines` just returns `quick_xml::Result<()>`. Do we want to add a `flamegraph::Result` type that it returns instead?